packages/suite/src/views/wallet/transactions/TransactionList/useFetchTransactions.ts

```diff
@@ -0,0 +1,211 @@
+        (
+            page: number,
+            options: {
+                recursive?: boolean;
+                noLoading?: boolean;
+            } = {},
+        ) => {
+            console.log('fetchPage:page', page, options);
+            synchronize(async () => {
+                setFetching(true);
+                await dispatch(
+                    fetchTransactionsPageThunk({
+                        accountKey: account.key,
+                        page,
+                        perPage,
+                        noLoading: Boolean(options.noLoading),
+                    }),
+                );
+                setFetching(false);
+            });
+        },
+        [account.key, dispatch, perPage, synchronize],
+    );
+
+    const fetchNext = useCallback(
+        () =>
+            synchronize(async () => {
+                if (fetchedAll) return;
+                setFetching(true);
+                await fetchCommon(pagesFetched + 1);
+                setFetching(false);
+                setPagesFetched(pagesFetched + 1);
+            }),
+        [synchronize, fetchCommon, pagesFetched, fetchedAll],
+    );
+
+    const fetchAll = useCallback(
+        () =>
+            synchronize(async () => {
+                if (fetchedAll) return;
+                setFetching(true);
+                await fetchCommon(pagesFetched + 1, {
+                    recursive: true,
+                });
+                setFetching(false);
+                setFetchedAll(true);
+            }),
+        [synchronize, fetchCommon, pagesFetched, fetchedAll],
+    );
+
+    return { fetchNext, pagesFetched, fetchPage, fetchAll, isFetching, fetchedAll };
+};
+
+export const useVisibleTransactionsRetriever = ({ account }: { account: Account }) => {
+    const allTransactions = useSelector(state =>
+        selectAccountTransactionsWithNulls(state, account.key),
+    );
+
+    const { fetchedAll, isFetching, pagesFetched, fetchPage } = useFetchTransactions(
+        account,
+        allTransactions,
+    );
+    const allAccountTransactions = useSelector(state =>
+        selectAccountTotalTransactions(state, account.key),
+    );
+    const transactionsIsLoading = useSelector(state =>
+        selectIsLoadingAccountTransactions(state, account.key),
+    );
+    const tokenDefinitions = useSelector(state =>
+        selectNetworkTokenDefinitions(state, account.symbol),
+    );
+
+    const visibleTransactions = useMemo(
+        () =>
+            tokenDefinitions
+                ? allTransactions.filter(
+                      transaction => !getIsPhishingTransaction(transaction, tokenDefinitions),
+                  )
+                : allTransactions,
+        [allTransactions, tokenDefinitions],
+    );
+
+    const perPage = getTxsPerPage(account.networkType);
+    const numberOfHidden = allTransactions.length - visibleTransactions.length;
+    const totalPossiblyVisible = allAccountTransactions - numberOfHidden;
+
+    useEffect(() => {
```

Comment:
This is obviously not idea, better solution would be to move this "fetching with condition" right to place where fetching is occuring - in the thunks, but that would require a lot of changes IMHO, this is simpler for this particular case.
I didn't want to be stuck in the complex thunks. Event hough I prefer it.